audio_alsa.c

```diff
@@ -560,10 +560,22 @@ static int actual_open_alsa_device(int do_auto_setup) {
     return ret;
   }
 
-  ret = snd_pcm_hw_params_set_channels(alsa_handle, alsa_params, 2);
+  int channels = 2;
+  if (config.playback_mode == ST_one_channel) {
+    channels = 1;
+  } else {
```

Comment:
I just don't understand the logic of this. Surely the default, if things fail, should be two channels, not one.

Reply:
At first I did a the manual config option, then I tried to find some APIs to find the number of channels instead.
I am not sure the preferred approach.
I would think getting rid of the config option would be better but I am not sure where to save the state that we are in the one channel mode.

But the logic is:
Check the manual config option is set then go right to one channel.
If the manual config option is not set, then test to see if 2 channels will work.
If not then and only then will it go to 1 channel.
1 channel is never the default.

The code before any changes will always fail on a one channel device and error out when it tries to set 2 channels.

If  I get rid of the config option the logic could be simpler.